svcb.go

```diff
@@ -531,11 +525,10 @@ func (*SVCBPort) len() int             { return 2 }
 func (s *SVCBPort) String() string     { return strconv.FormatUint(uint64(s.Port), 10) }
 func (s *SVCBPort) copy() SVCBKeyValue { return &SVCBPort{s.Port} }
 
-func (s *SVCBPort) unpack(b []byte) error {
-	if len(b) != 2 {
-		return errors.New("dns: svcbport: port length is not exactly 2 octets")
+func (s *SVCBPort) unpack(b *cryptobyte.String) error {
+	if !b.ReadUint16(&s.Port) {
+		return errUnpackOverflow
```

Comment:
regarding this errors: I think the est. pattern in this lib is: "bad.. ", with maybe slightly more context than just 'bad SVCB'